ducktape/command_line/parse_args.py

```diff
@@ -58,6 +58,12 @@ def create_ducktape_parser():
                         help="Upper bound on number of tests run simultaneously.")
     parser.add_argument("--repeat", action="store", type=int, default=1,
                         help="Use this flag to repeat all discovered tests the given number of times.")
+    parser.add_argument("--subsets", action="store", type=int, default=1,
```

Comment:
If I use your data and split it into 15 splits, the runtimes vary from 1710 sec to 2566 sec.
`
>>> for i in range(15): sum(itertools.islice(data, i, None, 15))
...
2566.3769999999995
2150.7529999999997
2244.764
2363.585
2320.717
2524.1629999999996
2092.3149999999996
2101.9869999999996
2063.3659999999995
2141.384
1718.5469999999998
1894.3680000000004
1710.8980000000001
1831.2150000000001
2200.502
`
I think that a completely automated solution of this problem is not clear at this point. I am inclined towards giving the end user as much control as possible and let them figure out how they want to do splits. (of course the user may choose to use data to generate splits - he may write a script to automate this split generation)

That said, if you feel that the Nth approach will work, let us go ahead with it and we can tackle the problem when we face it.